src/AggregateFunctions/SingleValueData.cpp

```diff
@@ -0,0 +1,1419 @@
+    return memory.get().setIfGreater(other.memory.get(), arena);
+}
+
+template <typename T>
+bool SingleValueDataNumeric<T>::setIfSmaller(const DB::IColumn & column, size_t row_num, DB::Arena * arena)
+{
+    return memory.get().setIfSmaller(column, row_num, arena);
+}
+
+template <typename T>
+bool SingleValueDataNumeric<T>::setIfGreater(const DB::IColumn & column, size_t row_num, DB::Arena * arena)
+{
+    return memory.get().setIfGreater(column, row_num, arena);
+}
+
+template <typename T>
+void SingleValueDataNumeric<T>::setSmallest(const DB::IColumn & column, size_t row_begin, size_t row_end, DB::Arena * arena)
+{
+    return memory.get().setSmallest(column, row_begin, row_end, arena);
+}
+
+template <typename T>
+void SingleValueDataNumeric<T>::setGreatest(const DB::IColumn & column, size_t row_begin, size_t row_end, DB::Arena * arena)
+{
+    return memory.get().setGreatest(column, row_begin, row_end, arena);
+}
+
+template <typename T>
+void SingleValueDataNumeric<T>::setSmallestNotNullIf(
+    const DB::IColumn & column,
+    const UInt8 * __restrict null_map,
+    const UInt8 * __restrict if_map,
+    size_t row_begin,
+    size_t row_end,
+    DB::Arena * arena)
+{
+    return memory.get().setSmallestNotNullIf(column, null_map, if_map, row_begin, row_end, arena);
+}
+
+template <typename T>
+void SingleValueDataNumeric<T>::setGreatestNotNullIf(
+    const DB::IColumn & column,
+    const UInt8 * __restrict null_map,
+    const UInt8 * __restrict if_map,
+    size_t row_begin,
+    size_t row_end,
+    DB::Arena * arena)
+{
+    return memory.get().setGreatestNotNullIf(column, null_map, if_map, row_begin, row_end, arena);
+}
+
+template <typename T>
+std::optional<size_t> SingleValueDataNumeric<T>::getSmallestIndex(const IColumn & column, size_t row_begin, size_t row_end) const
+{
+    return memory.get().getSmallestIndex(column, row_begin, row_end);
+}
+
+template <typename T>
+std::optional<size_t> SingleValueDataNumeric<T>::getGreatestIndex(const IColumn & column, size_t row_begin, size_t row_end) const
+{
+    return memory.get().getGreatestIndex(column, row_begin, row_end);
+}
+
+template <typename T>
+std::optional<size_t> SingleValueDataNumeric<T>::getSmallestIndexNotNullIf(
+    const IColumn & column, const UInt8 * __restrict null_map, const UInt8 * __restrict if_map, size_t row_begin, size_t row_end) const
+{
+    return memory.get().getSmallestIndexNotNullIf(column, null_map, if_map, row_begin, row_end);
+}
+
+template <typename T>
+std::optional<size_t> SingleValueDataNumeric<T>::getGreatestIndexNotNullIf(
+    const IColumn & column, const UInt8 * __restrict null_map, const UInt8 * __restrict if_map, size_t row_begin, size_t row_end) const
+{
+    return memory.get().getGreatestIndexNotNullIf(column, null_map, if_map, row_begin, row_end);
+}
+
+#define DISPATCH(TYPE) template struct SingleValueDataNumeric<TYPE>;
+
+FOR_SINGLE_VALUE_NUMERIC_TYPES(DISPATCH)
+#undef DISPATCH
+
+namespace
+{
+
+struct StringValueCompatibility
```

Comment:
This code is useful, I think it could be in header file for reusability(maybe ColumnString.h is a good place).

Reply:
Absolutely not. This code only exists to workaround of a bug in the past and we can't get rid of it without breaking compatibility. It should not be used anywhere else.

Although we welcome people creating their own forks and changing the code to fit whatever they need, what you are proposing in these comments is to make the code worse so that you have an easier integration and that's not ok.